### ISSUE TITLE
Why is the layout of $XDG_CONFIG_HOME/astropy different from $HOME/.astropy?

### ISSUE THREAD

**Issue by weaverba137** [astropy/astropy#6803]
If `$XDG_CONFIG_HOME` is set, astropy will look for configuration files in `$XDG_CONFIG_HOME/astropy/`.  However, it will **not** look for configuration files in `$XDG_CONFIG_HOME/astropy/config/`.  Similarly, if `$XDG_CACHE_HOME` is set, astropy will look for cache files in `$XDG_CACHE_HOME/astropy/`, not `$XDG_CACHE_HOME/astropy/cache`.  Why does the layout of `$XDG_CONFIG_HOME/astropy/` have to be different from the layout of `$HOME/.astropy/`, which contains `config/` and `cache/` directories?  Unless one knows that the layout is different, one would expect that `$XDG_CONFIG_HOME/astropy/config/` and `$XDG_CACHE_HOME/astropy/cache/` would be the directories to contain configuration and cache files.

**Comment by saimn** [comment]
Because XDG_CONFIG_HOME and XDG_CACHE_HOME are Linux standards for config and cache (usually `~/.config` and `~/.cache`). For me the current behavior makes perfect sense, if XDG_CONFIG_HOME is set it goes to `~/.config/astropy` (typically**) otherwise it is `~/.astropy/config` (and the same for cache).

** Actually if XDG_CONFIG_HOME is not set, it should default to `$HOME/.config`, so Astropy uses this in a diverted way, which is confusing, see also #6516 recently. Maybe it would have been better to have an `ASTROPY_CONFIG_DIR` variable instead to customize the `$HOME/.astropy/` paths directly, instead of trying to follow the xdg spec.

**Comment by weaverba137** [comment]
OK, maybe I should rephrase this: Why *doesn't* astropy use `~/.config/astropy` by default (on Unix-like systems)?

**Comment by Cadair** [comment]
@weaverba137 I came here because I wanted to know the answer to "Why doesn't astropy use `~/.config/astropy` by default (on Unix-like systems)?" Should we make this the title of the issue or something?

**Comment by weaverba137** [comment]
I would be happy to change the title.

**Comment by pllim** [comment]
I would say that this ship has sailed. Unless we do something drastic like #8245, too much overhead in refactoring (including answering issues from confused users on the behavior change) for little reward.

**Comment by weaverba137** [comment]
OK, but as @Cadair said:

> This is not the "correct" behaviour on a linux system as far as I am aware.

Can we at least *document* that we are *aware* that this is not correct behavior?

**Comment by Cadair** [comment]
We should absolutely do #8245 and I am -1 on closing this as wont-fix.

**Comment by pllim** [comment]
Feel free to open a PR for the desired documentation, as you already have an idea anyway.

As for #8245, it is not trivial. Maybe for 5.0 if someone is willing to invest the time and energy, but not before.

**Comment by Cadair** [comment]
I agree it's not easy, but this bug *should* be fixed which is why I don't want to close it. :smile:

**Comment by neutrinoceros** [comment]
#19575 might be our only chance to close this, but it currently does not. Please let me know it should.

**Comment by Cadair** [comment]
I'd vote a strong yes to fixing it if there's an opportunity

**Comment by neutrinoceros** [comment]
And just in case anyone is getting shaky legs from the current size of my PR, just a note that I already worked out a spec for how to fix this issue as well and I _don't_ expect it to be significantly more complicated than what I already have. It _is_ a different spec though.

**Comment by neutrinoceros** [comment]
and I think I just convinced myself of the overall superiority of this new spec while toying with it so the work is now actively underway in #19575
